datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/recovery/SdxUpgradeRecoveryService.java

```diff
@@ -1,81 +1,49 @@
 package com.sequenceiq.datalake.service.upgrade.recovery;
 
-import static com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.recovery.RecoveryStatus.NON_RECOVERABLE;
-
 import javax.inject.Inject;
 import javax.ws.rs.WebApplicationException;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
-import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
 import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
 import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.recovery.RecoveryValidationV4Response;
 import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
-import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
 import com.sequenceiq.cloudbreak.common.exception.WebApplicationExceptionMessageExtractor;
 import com.sequenceiq.cloudbreak.exception.CloudbreakApiException;
 import com.sequenceiq.cloudbreak.logger.MDCBuilder;
-import com.sequenceiq.cloudbreak.message.CloudbreakMessagesService;
 import com.sequenceiq.datalake.entity.SdxCluster;
 import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
-import com.sequenceiq.datalake.service.FreeipaService;
-import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.recovery.RecoveryService;
 import com.sequenceiq.flow.api.model.FlowIdentifier;
 import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
 import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
 import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
-import com.sequenceiq.sdx.api.model.SdxRecoveryType;
 
 @Component
-public class SdxUpgradeRecoveryService {
+public class SdxUpgradeRecoveryService implements RecoveryService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SdxUpgradeRecoveryService.class);
 
-    @Inject
-    private SdxService sdxService;
-
     @Inject
     private SdxReactorFlowManager sdxReactorFlowManager;
 
-    @Inject
-    private CloudbreakMessagesService messagesService;
-
-    @Inject
-    private FreeipaService freeipaService;
-
     @Inject
     private StackV4Endpoint stackV4Endpoint;
 
     @Inject
     private WebApplicationExceptionMessageExtractor exceptionMessageExtractor;
 
-    public SdxRecoveryResponse triggerRecovery(String userCrn, NameOrCrn clusterNameOrCrn, SdxRecoveryRequest recoverRequest) {
-        SdxCluster cluster = sdxService.getByNameOrCrn(userCrn, clusterNameOrCrn);
-        MDCBuilder.buildMdcContext(cluster);
-        return initSdxRecovery(recoverRequest, cluster);
-    }
+    public SdxRecoveryResponse triggerRecovery(SdxCluster sdxCluster, SdxRecoveryRequest recoverRequest) {
```

Comment:
you've explicitly removed validation from `triggerRecovery`, why is that? 
It was nice to make an extra validation to be sure recovery can be performed to cover if
- validation was not called explicitly
- something has changed since validation was called explicitly

Reply:
the validation is now explicitly in the selector service as a part of choosing which recovery service to use.